### PR TITLE
The concept of cookies in servlets

### LINKED ISSUE #4521: The concept of cookies in servlets
#### Proposed title of article
The concept of cookies in servlets

#### Proposed article introduction
Cookies are little pieces of data that the web server sends to the client in response to the client's request. To put it another way, these are the most basic ways for saving client state. Cookies are always kept on the computer of the client. They have a life expectancy and are deleted by the client browser when that life expectancy expires.

The Cookie class in the Servlet API is used to create cookies. The addCookie() method is used to add cookies to the response object. This method uses the HTTP response stream to transfer cookie data. To access the cookies that have been added to the response object, use the getCookies() method.

In this article, I'll go over everything you need to know about cookies so that both beginners and experienced Java programmers can use them effectively in their projects.
#### Key takeaways 
1. API for Cookies
2. How does Cookie work?
3. Various Cookie Types
4. Advantage and disadvantage of Cookies
5. Class and methods required for cookies
6. How to create, delete, and get cookies
7. Using an example to demonstrate Servlet Cookie
#### Article quality
There are several articles on the subject of cookies in javaScript, PHP, and other languages, but none on cookies in servlets. In addition, there is a view article in the section for servlets in Java. We'll go over the cookies concept in servlets in-depth in this article.
I've read every article about servlets and will make this one stand out from the rest. Furthermore, I will recommend that more article writers pay attention to the parts that are often overlooked, such as this. Thank you very much.

#### References
N/A

### PR DESCRIPTION
**Pull request by Evanslodoctor**
This fixes #4521

### CODE REVIEW

File: content/articles/The-concept-of-cookies-in-servlets/index.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+  - [Setting up lifespan for a cookie](#setting-up-lifespan-for-a-cookie)
+  - [Sending the cookies to the client](#sending-the-cookies-to-the-client)
+  - [Getting cookie from client request](#getting-cookie-from-client-request)
+  - [Deletting cookies](#deletting-cookies)
+- [Example to demonstrate Servlet Cookie](#example-to-demonstrate-servlet-cookie)
+- [Conclusion](#conclusion)
+
+### API for Cookies
+
+The `Cookie` class in the Servlet API is used to create cookies. Cookies are created using the Servlet API's 'Cookie' class. This class employs a variety of approaches to give various cookie activities. The `addCookie()` method, for example, is used to add cookies to the response object. It sends cookie data from a client to a server or from a server to a client using the `HTTP` response. The `getCookies()` method, on the other hand, is used to access the cookies that have been added to the response object.
```
Comment (paulodhiambo) [inline content/articles/The-concept-of-cookies-in-servlets/index.md:38]: Change "It sends cookie data from a client to a server or from a server to a client using the `HTTP` response." to "Then, it sends cookie data from a client to a server or server to a client using the `HTTP` response."

File: content/articles/The-concept-of-cookies-in-servlets/index.md [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+### How does Cookie work
+When a client makes a new request to a Web Server, the web server assigns a unique ID as a `cookie`, which is then sent back with the response to the client and saved in the client's browser.
```
Comment (paulodhiambo) [inline content/articles/The-concept-of-cookies-in-servlets/index.md:41]: Change "web server" to " webserver "

File: content/articles/The-concept-of-cookies-in-servlets/index.md [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+- Simplest technique: Cookies are the most basic method for keeping client states while connected to a web server.
+- Because cookies are stored in the client browser, they reduce the load on the server's memory.
+### Disadvantage 
+-  Cookies will only work if the user's web browser allows them; otherwise, they will not.
+- In a Cookie object, only textual data can be stored.
+
+### Cookie class and the methods required for cookies
+
+#### Cookie class
+ A `Cookie`, is a class that contains several cookie-related methods that allow you to use cookies.
```
Comment (paulodhiambo) [inline content/articles/The-concept-of-cookies-in-servlets/index.md:69]: Change to " A `Cookie` class contains several cookie-related methods that allow you to use cookies."

File: content/articles/The-concept-of-cookies-in-servlets/index.md [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+#### Methods of Cookie class
+- `void setMaxAge()`: For example, this method can be used to set Cookie's maximum age. The cookie lifetime can be specified when max-age is set in web applications.
+- `void setValue()`: When a new value is assigned to the cookie, this method is used.
+- `String getName()`: Returns the cookie's name, which cannot be modified after it has been created.
+- `String getValue()`: This method aids in obtaining the cookie's current value.
+- `int getVersion()`: Cookie's version number is returned by this method.
+- `void setVersion()`: Cookie's version number is set in this method.
+- ` int getMaxAge()`: This method is used to determine a cookie's maximum age.
+- `String getPath()`: The path from the cookie is obtained using this method.
+- `void setPath()`: This method is used to specify a cookie path. In a web application, for example, it is possible to specify a specific path for the cookies to be saved.
```
Comment (paulodhiambo) [inline content/articles/The-concept-of-cookies-in-servlets/index.md:79]: Change to "- `void setPath()`: This method is used to specify a cookie path. For example, in a web application, it is possible to specify a specific path for the cookies to be saved."

File: content/articles/The-concept-of-cookies-in-servlets/index.md [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+        Enter subject <input type="text" name="subject" required><br>
+        <input type="submit" value="Click me">
+    </form>
+</body>
+
+</html>
+```
+
+#### Step 3: Creating a ProcessOneServlet Servlet page
+By right-clicking on the`Example to demonstrate cookies in servlet` project selecting New Servlet, you can code a Servlet page as shown below. Provide your Servlet class name. I will call it `ProcessOneServlet`.
```
Comment (paulodhiambo) [inline content/articles/The-concept-of-cookies-in-servlets/index.md:150]: Change to "You can code a Servlet page as shown below by right-clicking on the`Example to demonstrate cookies in servlet` project selecting New Servlet. First, provide your Servlet class name. I will call it `ProcessOneServlet`."

File: content/articles/The-concept-of-cookies-in-servlets/index.md [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+		response.addCookie(object);
+		object2.setMaxAge(365*24*60*60);
+		response.addCookie(object2);
+		RequestDispatcher req=request.getRequestDispatcher("ReadingValuesFromCookies");
+		req.forward(request, response);
+	}
+
+}
+```
+#### Step 3: Creating a ReadingValuesFromCookies Servlet page
```
Comment (paulodhiambo) [inline content/articles/The-concept-of-cookies-in-servlets/index.md:187]: This should be #### Step 4

File: content/articles/The-concept-of-cookies-in-servlets/index.md [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+By right-clicking on the`Example to demonstrate cookies in servlet` project selecting New Servlet, you can code a Servlet page as shown below. Provide your Servlet class name. I will call it `ReadingValuesFromCookies`.
```
Comment (paulodhiambo) [inline content/articles/The-concept-of-cookies-in-servlets/index.md:188]: Change to "You can code a Servlet page as shown below by right-clicking on the`Example to demonstrate cookies in servlet` project selecting New Servlet. Provide your Servlet class name. I will call it `ReadingValuesFromCookies`."

**Review by paulodhiambo — CHANGES_REQUESTED** [review]
Hello @Evanslodoctor make the requested changes. I'll proceed to Tech review.

**Comment by Evanslodoctor** [comment]
> Hello @Evanslodoctor make the requested changes. I'll proceed to Tech review.

Hello @paulodhiambo? I'm done.

**Comment by WanjaMIKE** [comment]
Plagiarism check is done. Could you revisit some of the highlighted areas in the attached pdf. Thanks
[Uploading New Scan 5_33 PM Copyleaks Report.pdf…]()

**Comment by Evanslodoctor** [comment]
> Plagiarism check is done. Could you revisit some of the highlighted areas in the attached pdf. Thanks Uploading New Scan 5_33 PM Copyleaks Report.pdf…

@WanjaMIKE , how are you? This document is not available in PDF format. Could you please send it again? When I try to open it, it takes me back to this page.

**Comment by WanjaMIKE** [comment]
Sure, I will

**Comment by Evanslodoctor** [comment]
> Sure, I will

Waiting.

**Comment by hectorkambow** [comment]
Pinged Mike via slack as well 👍

**Comment by WanjaMIKE** [comment]
Please check the pdf. Thanks
[The concept of cookies in servlets #4832 Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/7524692/The.concept.of.cookies.in.servlets.4832.Copyleaks.Report.pdf)

**Comment by Evanslodoctor** [comment]
> Please check the pdf. Thanks [The concept of cookies in servlets #4832 Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/7524692/The.concept.of.cookies.in.servlets.4832.Copyleaks.Report.pdf)

I am done. The remaining plagiarisms is for java imports and functions which are predefined. eg.,
import java.io.IOException;
import java.io.PrintWriter;
import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.Cookie;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
@WebServlet("/ReadingValuesFromCookies")
public class ReadingValuesFromCookies extends HttpServlet


thank you.

**Comment by paulodhiambo** [comment]
> > Please check the pdf. Thanks [The concept of cookies in servlets #4832 Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/7524692/The.concept.of.cookies.in.servlets.4832.Copyleaks.Report.pdf)
> 
> I am done. The remaining plagiarisms is for java imports and functions which are predefined. eg., import java.io.IOException; import java.io.PrintWriter; import javax.servlet.ServletException; import javax.servlet.annotation.WebServlet; import javax.servlet.http.Cookie; import javax.servlet.http.HttpServlet; import javax.servlet.http.HttpServletRequest; import javax.servlet.http.HttpServletResponse; @WebServlet("/ReadingValuesFromCookies") public class ReadingValuesFromCookies extends HttpServlet
> 
> thank you.

@Evanslodoctor please don't include the import statements unless necessary to keep the article clean.

**Comment by hectorkambow** [comment]
@paulodhiambo 

Any update on this PR? 

Keeping in mind that any extra revisions should occur outside of the queue as to not increase the back log.

**Comment by ahmadmardeni1** [comment]
Plagiarism check done ✅

**Comment by hectorkambow** [comment]
@ahmadmardeni1  

Why did we run two plagiarism checks on this ?

**Comment by hectorkambow** [comment]
@paulodhiambo can we get a status update on this PR?

**Comment by paulodhiambo** [comment]
Good article @Evanslodoctor, please change your article folder name from "The-concept-of-cookies-in-servlets" to "the-concept-of-cookies-in-servlets"

**Pull request merged**